docs/learn/guides/file-integrations/file-integration-using-ftp/README.md

```diff
@@ -0,0 +1,198 @@
+
+### Create the project structure
+
+Ballerina is a complete programming language that supports custom project structures. Use the following package structure for this guide.
+```
+file-integration-using-ftp
+    ├── src
+       └── ftp_listener
+           └── ftp_listener.bal
+```
+
+Create the Ballerina project `file-integration-using-ftp` and add the `ftp_listener` module using the below commands. 
+
+```bash
+    $ ballerina new file-integration-using-ftp
+    $ cd file-integration-using-ftp
+    $ ballerina add ftp_listener
+```
+
+The above package structure will be created for you. Create the `ftp_listener.bal` file in the Ballerina module.
+
+### Developing the FTP listener service
+
+Let's start implementation by importing the WSO2 FTP Connector in the `ftp_listener.bal` file you just created.
+
+```ballerina
+import wso2/ftp;
+```
+
+Then add the configuration for file type and file location to listen to.
+
+ ```ballerina
+type Config record {
+    string fileNamePattern;
+    string filePath;
+};
+
+Config conf = {
+    fileNamePattern: config:getAsString("FTP_FILE_NAME_PATTERN"),
+    filePath: config:getAsString("FTP_LISTENER_PATH")
+};
+```
+
+Create an FTP listener instance by defining the configuration.
+
+```ballerina
+listener ftp:Listener dataFileListener = new({
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+        basicAuth: {
+            username: config:getAsString("FTP_USERNAME"),
+            password: config:getAsString("FTP_PASSWORD")
+        }
+    },
+    path: conf.filePath,
+    fileNamePattern: conf.fileNamePattern
+    pollingInterval: config:getAsInt("FTP_POLLING_INTERVAL"),
+});
+```
+
+Then implement the FTP client endpoint to do the file processing.
+     
+```ballerina
+ftp:ClientEndpointConfig ftpConfig = {
+    protocol: ftp:FTP,
+    host: config:getAsString("FTP_HOST"),
+    port: config:getAsInt("FTP_LISTENER_PORT"),
+    secureSocket: {
+     basicAuth: {
+         username: config:getAsString("FTP_USERNAME"),
+         password: config:getAsString("FTP_PASSWORD")
+     }
+    }
+};
+
+ftp:Client ftpClient = new(ftpConfig);
+```
+
+Create the service to listen to the FTP server. 
+
+```ballerina
+service dataFileService on dataFileListener {
+    resource function processDataFile(ftp:WatchEvent fileEvent) {
+
+        foreach ftp:FileInfo file in fileEvent.addedFiles {
+            log:printInfo("Added file path: " + file.path);
+            processNewFile(file.path);
+        }
+        foreach string file in fileEvent.deletedFiles {
+            log:printInfo("Deleted file path: " + file);
+            processDeletedFile(file.path);
+        }
+    }
+}
+```
+
+Declare a map to store file names and sizes.
+
+```ballerina
+map<int> fileMap = {};
+```
+
+Now, implement the processing of added and deleted files.
```

Comment:
Add some details on what exactly is happening when processing.

Reply:
Added.